MapboxCoreNavigation/RouteController.swift

```diff
@@ -360,6 +359,10 @@ extension RouteController: CLLocationManagerDelegate {
         
         self.lastRerouteLocation = location
         
+        if let accessToken = routeProgress.route.accessToken, let apiEndpoint = routeProgress.route.apiEndpoint, let host = apiEndpoint.host {
+            directions = Directions(accessToken: accessToken, host: host)
+        }
+        
         routeTask = directions.calculate(options, completionHandler: { [weak self] (waypoints, routes, error) in
```

Comment:
This is the only thing we were ever using the `directions` property for. I think we could get away with removing the `directions` property and the `directions` argument in the initializer, in favor of the new code above plus a fallback to `Directions.shared`. Do you think there would be any downside to that?